include/rmm/detail/aligned.hpp

```diff
@@ -125,6 +125,7 @@ namespace rmm::detail {
  * from `alloc`.
  *
  * If `alignment` is not a power of 2, behavior is undefined.
+ * If `Alloc` does not allocate host-accessible memory, behavior is undefined.
```

Comment:
**question**: Should this restriction on use of this function be called out more obviously? It feels like I have to be paying a lot of attention to know that passing `[](std::size_t bytes) { void *ptr; cudaMalloc(&ptr, bytes); return ptr; }` is now invalid usage.

Reply:
Let's rename the functions.

Reply:
Renamed to aligned_host_allocate/deallocate and fixed up the docs a bit more.

Reply:
Thanks.

Reply:
> is now invalid usage.

BTW, this has always been invalid. This code is/was only used for host allocation. The reason is that we didn't support alignment parameters previously, and since CUDA always aligns to 256 bytes, we only have to pad all allocation sizes to a multiple of that when suballocating device memory.